packages/core/CalendarManager.ts

```diff
@@ -266,7 +266,7 @@ export const updateEvent = async (
           })
       : undefined;
 
-  if (Array.isArray(updatedResult)) {
+  if (Array.isArray(updatedResult) && updatedResult?.additionalInfo) {
     calWarnings = updatedResult.flatMap((res) => res.additionalInfo.calWarnings ?? []);
   } else {
     calWarnings = updatedResult?.additionalInfo.calWarnings;
```

Comment:
```suggestion
    calWarnings = updatedResult?.additionalInfo?.calWarnings;
```

Reply:
Ignore my comments, without knowing exactly what calWarnings is when it is not an array I can't really provide the right suggestions - but it sounds to me like it should always be an array.